pkg/destroy/azure/azure.go

```diff
@@ -232,6 +236,10 @@ func deletePublicRecords(ctx context.Context, dnsClient dns.ZonesClient, records
 
 	privateZonesPage, err := privateDNSClient.ListByResourceGroup(ctx, rgName, to.Int32Ptr(100))
 	if err != nil {
+		if privateZonesPage.Response().IsHTTPStatus(http.StatusNotFound) {
+			logger.Debug("already deleted")
+			return nil
```

Comment:
```suggestion
			return utilerrors.NewAggregate(errs)
```
otherwise any errors collected earlier in the function will be swallowed

Reply:
Nice catch - addressed